File: agent-driven-architecture-main/examples/typescript/test/gate/fixtures/compliant/C16/src/spine/replay/replay.ts
// COMPLIANT: the attributed list is handed WHOLE to the admission rule, and what
// this file names is the value `admit` handed back.
//
// CONSTRUCTING an attribution stays legal wherever the fold runs — the fold is
// the only place holding both halves — and so does PROSE: this very comment says
// that an arm emitted an effect, and a rule that fired on it would be the
// nuisance 15.2 warns about. Only a member READ is denied.
export const derive = (admit, licences, produced) => admit(licences, produced);
export const attribute = (from, emitted) => ({ from, emitted });
export const named = (from, emitted, make) => make({ from: from, emitted: emitted });

File: agent-driven-architecture-main/examples/kotlin/src/test/fixtures/konsist/compliant/C16/spine/replay/Replay.kt
// COMPLIANT: the attributed list is handed WHOLE to the admission rule, and what
// this file names is the value `admit` handed back.
//
// FOUR SHAPES, each proven ACCEPTED, because the rule this replaces rejected two of
// them: a POSITIONAL construction, a NAMED-ARGUMENT construction, a declaration's
// own KDoc using the word in prose, and a `List<Attributed>` handed whole to the
// rule. CONSTRUCTING an attribution stays legal wherever the fold runs — only
// OPENING one is the admission rule's.

package adr.spine.replay

import adr.contract.Effect
import adr.contract.ToolResult
import adr.spine.pure.Admission
import adr.spine.pure.Attributed

class Derive {

    /** The decision-log lines the triage arm emitted, handed whole to the rule. */
    fun perform(admission: Admission, produced: List<Attributed>): List<Effect> =
        admission.admit(produced)

    /** Positional. What every fold arm writes. */
    fun attribute(from: ToolResult, effect: Effect): Attributed = Attributed(from, effect)

    /** Named-argument. The same construction, spelled the way a reader prefers. */
    fun named(from: ToolResult, effect: Effect): Attributed =
        Attributed(from = from, emitted = effect)
}
